src/components/TagContainer.ts

```diff
@@ -0,0 +1,237 @@
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+
+            this.subscriptionHandles.push(mx.data.subscribe({
+                attr: this.tagAttribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+        }
+    }
+
+    private handleSubscriptions() {
+        this.setState({
+            newTag: this.getValue(this.tagAttribute, this.props.mxObject) as string
+        });
+    }
+
+    private lazyLoadTags() {
+        if (this.props.mxObject) {
+            this.fetchCurrentTags(this.props.mxObject);
+        }
+    }
+
+    private fetchCurrentTags(mxObject: mendix.lib.MxObject) {
+        const { tagConstraint } = this.props;
+        const constraint = tagConstraint
+            ? tagConstraint.replace(/\[\%CurrentObject\%\]/gi, mxObject.getGuid())
+            : "";
+        const XPath = "//" + this.tagEntity + constraint;
+        mx.data.get({
+            callback: object => {
+                const suggestionList = object.map(tagObject => ({
+                    value: tagObject.get(this.tagAttribute) as string
+                }));
+                this.setState({ suggestions: suggestionList.map(suggestion => suggestion.value) });
+                this.processTags(object);
+            },
+            error: error =>
+                window.mx.ui.error(`An error occurred while retrieving tags via XPath (${this.tagEntity}): ${error}`),
+            xpath: XPath
+        });
+    }
+
+    private processTags(tagObjects: any[]) {
+        const currentTagObjects: mendix.lib.MxObject[] = [];
+        const referenceTags = this.props.mxObject.get(this.referenceAttribute) as string[];
+        tagObjects.map(object => {
+            referenceTags.map(reference => {
+                if (reference === object.getGuid()) {
+                    currentTagObjects.push(object);
+                }
+            });
+        });
+        const tagData = currentTagObjects.map(tagObject => ({ value: tagObject.get(this.tagAttribute) as string }));
+        this.setState({
+            tagList: tagData.map(tag => tag.value)
+        });
+    }
+
+    private createTag(tag: string) {
+        const { afterCreateMicroflow, mxObject } = this.props;
+        mx.data.create({
+            callback: object => {
+                object.set(this.tagAttribute, tag);
+                mx.data.commit({
+                    callback: () => {
+                        mxObject.addReference(this.referenceAttribute, object.getGuid());
+                        this.saveTag(mxObject);
+
+                        if (afterCreateMicroflow) {
+                            this.executeAction(afterCreateMicroflow, object.getGuid());
+                        }
+                    },
+                    error: error => window.mx.ui.error("Error occurred attempting to commit: " + error),
```

Comment:
if you catch error, never add error, the error is an object take `error.message`

Reply:
done